nsxt/resource_nsxt_policy_site.go

```diff
@@ -0,0 +1,267 @@
+						},
+						"site_uuid": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "id of Site",
+						},
+						"thumbprint": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Thumbprint of Enforcement Point",
+						},
+						"username": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Username",
+						},
+					},
+				},
+			},
+			"site_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Persistent Site Type",
+				ValidateFunc: validation.StringInSlice(siteTypeValues, false),
+			},
+		},
+	}
+}
+
+func getSiteFromSchema(d *schema.ResourceData) gm_model.Site {
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	var tags []gm_model.Tag
+	tagListIntface := d.Get("tag")
+	if tagListIntface != nil {
+		tagList := tagListIntface.(*schema.Set).List()
+		for _, t := range tagList {
+			elem := t.(map[string]interface{})
+			scope := elem["scope"].(string)
+			tag := elem["tag"].(string)
+			tags = append(tags, gm_model.Tag{Scope: &scope, Tag: &tag})
+		}
+	}
+	failIfRtepMisconfigured := d.Get("fail_if_rtep_misconfigured").(bool)
+	failIfRttExceeded := d.Get("fail_if_rtt_exceeded").(bool)
+	maximumRtt := int64(d.Get("maximum_rtt").(int))
+
+	var siteConnectionInfos []gm_model.SiteNodeConnectionInfo
+	siteConnectionInfoList := d.Get("site_connection_info").([]interface{})
+	for _, sci := range siteConnectionInfoList {
+		elem := sci.(map[string]interface{})
+		fqdn := elem["fqdn"].(string)
+		password := elem["password"].(string)
+		siteUUID := elem["site_uuid"].(string)
+		thumbprint := elem["thumbprint"].(string)
+		username := elem["username"].(string)
+
+		siteConnectionInfo := gm_model.SiteNodeConnectionInfo{
+			Fqdn:       &fqdn,
+			Password:   &password,
+			SiteUuid:   &siteUUID,
+			Thumbprint: &thumbprint,
+			Username:   &username,
+		}
+		siteConnectionInfos = append(siteConnectionInfos, siteConnectionInfo)
+	}
+	siteType := d.Get("site_type").(string)
+
+	return gm_model.Site{
+		DisplayName:             &displayName,
+		Description:             &description,
+		Tags:                    tags,
+		FailIfRtepMisconfigured: &failIfRtepMisconfigured,
+		FailIfRttExceeded:       &failIfRttExceeded,
+		MaximumRtt:              &maximumRtt,
+		SiteConnectionInfo:      siteConnectionInfos,
+		SiteType:                &siteType,
+	}
+}
+
+func resourceNsxtPolicySiteExists(id string, connector client.Connector, isGlobal bool) (bool, error) {
+	client := global_infra.NewSitesClient(connector)
+	_, err := client.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func resourceNsxtPolicySiteCreate(d *schema.ResourceData, m interface{}) error {
+	id, err := getOrGenerateID(d, m, resourceNsxtPolicySiteExists)
```

Comment:
can we add here: 
        if !isPolicyGlobalManager(m) {
                return globalManagerOnlyError()
        }